receiver/receiverhelper/scrapercontroller.go

```diff
@@ -58,63 +54,73 @@ func AddMetricsScraper(scraper MetricsScraper) ScraperControllerOption {
 // metrics will be passed to the next consumer.
 func AddResourceMetricsScraper(scrape ResourceMetricsScraper) ScraperControllerOption {
 	return func(o *scraperController) {
-		o.metricScrapers = append(o.metricScrapers, scrape)
+		o.resourceMetricScrapers = append(o.resourceMetricScrapers, scrape)
 	}
 }
 
 type scraperController struct {
-	defaultCollectionInterval time.Duration
-	nextConsumer              consumer.MetricsConsumer
+	collectionInterval time.Duration
+	nextConsumer       consumer.MetricsConsumer
 
-	metricScrapers []ResourceMetricsScraper
-	done           chan struct{}
+	metricsScrapers        *multiMetricScraper
+	resourceMetricScrapers []ResourceMetricsScraper
+	done                   chan struct{}
 }
 
 // NewScraperControllerReceiver creates a Receiver with the configured options, that can control multiple scrapers.
-func NewScraperControllerReceiver(_ configmodels.Receiver, nextConsumer consumer.MetricsConsumer, options ...ScraperControllerOption) (component.Receiver, error) {
+func NewScraperControllerReceiver(cfg ScraperControllerSettings, nextConsumer consumer.MetricsConsumer, options ...ScraperControllerOption) (component.Receiver, error) {
 	if nextConsumer == nil {
 		return nil, componenterror.ErrNilNextConsumer
 	}
 
-	mr := &scraperController{
-		defaultCollectionInterval: time.Minute,
-		nextConsumer:              nextConsumer,
-		done:                      make(chan struct{}),
+	sc := &scraperController{
+		collectionInterval: time.Minute,
+		nextConsumer:       nextConsumer,
+		metricsScrapers:    &multiMetricScraper{},
+		done:               make(chan struct{}),
+	}
+
+	if cfg.CollectionIntervalVal > 0 {
```

Comment:
You should return error if what is passed is 0. Maybe add a DefaultControllerSettings that return a default value.

Reply:
Added